internal/provider/provider.go

```diff
@@ -20,6 +23,15 @@ type Provider struct {
 type ProviderModel struct {
 }
 
+type ProviderOpts struct {
+	ropts    []remote.Option
+	keychain authn.Keychain
+}
+
+func (p *ProviderOpts) withContext(ctx context.Context) []remote.Option {
+	return append([]remote.Option{remote.WithContext(ctx)}, p.ropts...)
```

Comment:
Should this add in `remote.WithKeychain(p.keychain)`?

Reply:
That's in the original `p.ropts`, the only reason we keep `p.keychain` around is because `Verify` takes both a keychain and opts.